x/distribution/types/store.go

```diff
@@ -1,7 +1,26 @@
 package types
 
-const ()
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
 
-// store: per-validator deccoins, for bonded set
-// store: per-validator accumulated commission
-// store: storage records from withdrawals (fees / power)
+// historical rewards for a validator
+type ValidatorHistoricalRewards struct {
+	AccumulatedFees sdk.DecCoins `json:"accumulated_fees"`
+	TotalPower      sdk.Dec      `json:"total_power"`
```

Comment:
Is this taking an optimization to avoid truncation error, where the stored field isn't `accumulatedFees / that_periods_power`, but instead `Accumulated Fees, Total Power`? Its some extra space overhead, but may be worth it for lower error!

Reply:
I originally did this, then ditched it, but might try again.

Reply:
Ah okay! I don't expect it to be that important if were truncating, but probably worth us working out with pen and paper